### PR TITLE
Add UpdateFeatureGroup related APIs in sample notebook

### PR DESCRIPTION
**Pull request by verayu43**
*Issue #, if available:*
Add new launched APIs -- UpdateFeatureGroup.

*Description of changes:*
Add new launched APIs -- UpdateFeatureGroup.

*Testing done:*
Tested in the JupyterLab.

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [x] I have read the [CONTRIBUTING](https://github.com/aws/amazon-sagemaker-examples/blob/master/CONTRIBUTING.md) doc and adhered to the example notebook best practices
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-sagemaker-examples/blob/master/README.md)
- [x] I have tested my notebook(s) and ensured it runs end-to-end
- [x] I have linted my notebook(s) and code using `tox -e black-format,black-nb-format`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: sagemaker-featurestore/feature_store_introduction.ipynb [hunk 1]
```diff
@@ -310,6 +310,63 @@
...
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sagemaker.feature_store.feature_definition import (\n",
+    "    FractionalFeatureDefinition,\n",
+    "    IntegralFeatureDefinition,\n",
+    "    StringFeatureDefinition,\n",
+    "    FeatureTypeEnum,\n",
```
Comment (imingtsou) [inline sagemaker-featurestore/feature_store_introduction.ipynb:332]: Do we need all types of FeatureDefinition in import?

File: sagemaker-featurestore/feature_store_introduction.ipynb [hunk 2]
```diff
@@ -473,6 +530,51 @@
...
+    "        }        \n",
+    "    ]\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Use the `GetRecord` operation again to check the records that have been ingested to the updated customers_feature_group."
```
Comment (imingtsou) [inline sagemaker-featurestore/feature_store_introduction.ipynb:575]: Where is the verification?

Comment (verayu43) [reply sagemaker-featurestore/feature_store_introduction.ipynb:575]: I will add another cell using GetRecord to validate the above PutRecord is made successful.

File: sagemaker-featurestore/feature_store_introduction.ipynb (hunk 1 above)
Comment (verayu43) [reply sagemaker-featurestore/feature_store_introduction.ipynb:332]: Not required for all types here, will revise to only import StringFeatureDefinition.

File: sagemaker-featurestore/feature_store_introduction.ipynb [hunk 3]
```diff
@@ -473,6 +526,62 @@
     "all_records"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If we want to load data for the added features into the updated customers_feature_group, use `PutRecord` operation."
+   ]
+  },
```
Comment (alex-tang) [inline sagemaker-featurestore/feature_store_introduction.ipynb:535]: What do you think about moving the schema update code down here right before ingesting records with the new columns? It doesn't seem very useful to create a feature group, immediately update its schema, break up ingestion in to 2 consecutive parts (old schema and new schema).

File: sagemaker-featurestore/feature_store_introduction.ipynb [hunk 4]
```diff
@@ -473,6 +526,62 @@
...
+    "    FeatureGroupName=customers_feature_group_name,\n",
+    "    Record=[\n",
+    "        {\"FeatureName\": \"customer_id\", \"ValueAsString\": \"573291\"},\n",
+    "        {\"FeatureName\": \"EventTime\", \"ValueAsString\": str(current_time_sec)},\n",
+    "        {\"FeatureName\": \"email\", \"ValueAsString\": \"john.lee@gmail.com\"},\n",
+    "        {\"FeatureName\": \"name\", \"ValueAsString\": \"John Lee\"},\n",
+    "    ],\n",
+    ")"
+   ]
+  },
```
[the remark this answers is not in the document]
Comment (verayu43) [reply sagemaker-featurestore/feature_store_introduction.ipynb]: yes, I revised in the new version using .ingest.

File: sagemaker-featurestore/feature_store_introduction.ipynb (hunk 3 above)
Comment (verayu43) [reply sagemaker-featurestore/feature_store_introduction.ipynb]: agree. revision made in the new version.

**Review by alex-tang — APPROVED** [review]

File: sagemaker-featurestore/data/feature_store_introduction_customer_updated.csv [hunk 5]
```diff
@@ -0,0 +1,5 @@
+customer_id,city_code,state_code,country_code,email,name
+573291,1,49,2,john.lee@gmail.com,John Lee
+109382,2,40,2,olivequil@gmail.com,Olive Quil
+828400,3,31,2,liz.knee@gmail.com,Liz Knee
+124013,4,5,2,eileenbook@gmail.com,Elieen Book
```
Comment (jkroll-aws) [inline sagemaker-featurestore/data/feature_store_introduction_customer_updated.csv:5]: Spelling nit: `Elieen` -> `Eileen`?

File: sagemaker-featurestore/feature_store_introduction.ipynb [hunk 6]
```diff
@@ -473,6 +473,152 @@
     "all_records"
```
Comment (jkroll-aws) [inline sagemaker-featurestore/feature_store_introduction.ipynb:473]: <pre class="ql-syntax" spellcheck="false">Line #4.            print("Waiting for Feature Group to be updated")
</pre><p>Branding nit - should it be "feature group", "FeatureGroup", or "Feature Group"? Use one consistently.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3515/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='46'/>

Comment (jkroll-aws) [inline sagemaker-featurestore/feature_store_introduction.ipynb:473]: <p>Formatting nit: backticks around <code>customers_feature_group</code>?  <span class="ql-cursor">﻿</span></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3515/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='54'/>

Comment (verayu43) [reply sagemaker-featurestore/feature_store_introduction.ipynb:473]: <p>thx! fixed in new version.</p>

Comment (verayu43) [reply sagemaker-featurestore/feature_store_introduction.ipynb:473]: <p>added <code></code> around customers_feature_group in new version.</p>

File: sagemaker-featurestore/data/feature_store_introduction_customer_updated.csv (hunk 5 above)
Comment (verayu43) [reply sagemaker-featurestore/data/feature_store_introduction_customer_updated.csv]: thx for catching it, fixed in new version.

**Review by jkroll-aws — APPROVED** [review]

**Pull request merged**